File: src/domain/comic.ts
/**
 * Comic-domain DTOs rendered by the frontend.
 *
 * Novel and comic remain separate domains, but their summaries, catalogue
 * entries, reading positions, and reader payloads live here instead of in
 * transport adapters.
 */
import type { ReadPosition } from "./readPosition";

export interface ComicSummary {
  id: string;
  /** Official numeric volume ID, populated only for bookshelf entries. */
  bookId?: string | null;
  title: string;
  coverUrl: string | null;
  author: string | null;
}

export interface ComicChapterSummary {
  id: string;
  title: string;
  sequence: number;
  pageCount: number;
}

export interface ComicBook {
  id: string;
  title: string;
  readPosition: ReadPosition | null;
  chapters: ComicChapterSummary[];
}

export interface ComicSeriesDetail extends ComicSummary {
  description: string | null;
  genre: string[];
  status: string;
  books: ComicBook[];
}

export interface ComicChapterPageBatch {
  chapterId: string;
  startIndex: number;
  pageCount: number;
  pageUrls: string[];
  readPosition: ReadPosition | null;
}

File: src-tauri/src/dto/comic.rs
use crate::dto::common::ReadPosition;
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicSummary {
    pub id: String,
    /// 官方书架中的数值分卷 ID；仅书架项目具备该值。
    pub book_id: Option<String>,
    pub title: String,
    pub cover_url: Option<String>,
    pub author: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicChapterSummary {
    pub id: String,
    pub title: String,
    pub sequence: i64,
    pub page_count: i64,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicBook {
    pub id: String,
    pub title: String,
    pub read_position: Option<ReadPosition>,
    pub chapters: Vec<ComicChapterSummary>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicSeriesDetail {
    #[serde(flatten)]
    pub summary: ComicSummary,
    pub description: Option<String>,
    pub genre: Vec<String>,
    pub status: String,
    pub books: Vec<ComicBook>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicChapterPageBatch {
    pub chapter_id: String,
    pub start_index: i64,
    pub page_count: i64,
    pub page_urls: Vec<String>,
    pub read_position: Option<ReadPosition>,
}
